components/RegisterISCN.vue

```diff
@@ -0,0 +1,472 @@
+<template>
+  <div class="flex flex-col gap-6">
+    <!-- Basic Info -->
+    <UFormGroup label="Type" class="flex-1">
+      <UInput v-model="iscnData.type" placeholder="Type" />
+    </UFormGroup>
+
+    <UFormGroup label="Title" class="flex-1" required>
+      <UInput v-model="iscnData.title" placeholder="Enter ISCN title" />
+    </UFormGroup>
+
+    <UFormGroup label="Description" class="flex-1" required>
+      <UTextarea
+        v-model="iscnData.description"
+        placeholder="Enter ISCN description"
+        autoresize
+      />
+    </UFormGroup>
+
+    <div class="grid grid-cols-3 gap-4">
+      <UFormGroup label="ISBN">
+        <UInput v-model="iscnData.isbn" placeholder="Enter ISBN" />
+      </UFormGroup>
+
+      <UFormGroup label="Publisher">
+        <UInput
+          v-model="iscnData.publisher"
+          placeholder="Enter publisher name"
+        />
+      </UFormGroup>
+
+      <UFormGroup label="Publication Date">
+        <UInput
+          v-model="iscnData.publicationDate"
+          type="date"
+          placeholder="Select date"
+        />
+      </UFormGroup>
+
+      <UFormGroup label="Language">
+        <USelect
+          v-model="iscnData.language"
+          :options="languageOptions"
+          placeholder="Select language"
+        />
+      </UFormGroup>
+
+      <UFormGroup label="書訊">
+        <UInput
+          v-model="iscnData.bookInfoUrl"
+          placeholder="Enter book info URL"
+        />
+      </UFormGroup>
+    </div>
+
+    <!-- Author Info -->
+    <div class="grid grid-cols-2 gap-4">
+      <UFormGroup label="Author Name" required>
+        <UInput
+          v-model="iscnData.author.name"
+          placeholder="Enter author name"
+        />
+      </UFormGroup>
+
+      <UFormGroup label="Author Description">
+        <UTextarea
+          v-model="iscnData.author.description"
+          placeholder="Enter author description"
+          autoresize
+        />
+      </UFormGroup>
+    </div>
+
+    <UFormGroup label="Author Avatar URL">
+      <UInput
+        v-model="iscnData.author.url"
+        placeholder="Enter author avatar URL"
+      />
+    </UFormGroup>
```

Comment:
Avatar should be `iscnData.author.image`, but maybe we don't want to add `Author Avatar URL` in this PR

Reply:
[🔥 Remove unused code](https://github.com/likecoin/nft-book-press/pull/292/commits/adbc0abe592b3005e38bdaa74918f8e57a5dc61c)